graylog2-web-interface/src/views/components/visualizations/hooks/useLayoutExtendedSettings.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { useCallback, useMemo } from 'react';
+import type { Layout } from 'plotly.js';
+
+import {
+  generateDomain,
+  generateLayouts,
+  generateMappersForYAxis,
+} from 'views/components/visualizations/utils/chartLayoytGenerators';
+import type AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import type { BarMode } from 'views/logic/aggregationbuilder/visualizations/BarVisualizationConfig';
+import type { ChartDefinition } from 'views/components/visualizations/ChartData';
+import useWidgetUnits from 'views/components/visualizations/hooks/useWidgetUnits';
+import useFeature from 'hooks/useFeature';
+import { UNIT_FEATURE_FLAG } from 'views/components/visualizations/Constants';
+
+const useLayoutExtendedSettings = ({ config, barmode, chartData }: {
```

Comment:
The naming is not great here, it should describe what the hook is actually doing.